internal/logger/logger_test.go

```diff
@@ -42,53 +43,57 @@ var _ = Describe("Logger", func() {
 
 	It("can log that it is skipping an instance", func() {
 		result := captureStdout(func() {
-			l.SkippingInstance("my-instance", "fake-guid", true, "create", "failed")
+			l.SkippingInstance(fullInstance("my-instance", "fake-guid", true, "create", "failed"))
 		})
 		Expect(result).To(MatchRegexp(timestampRegexp + `: skipping instance: "my-instance" guid: "fake-guid" Upgrade Available: true Last Operation: Type: "create" State: "failed"\n`))
 	})
 
 	It("can log the start of an upgrade", func() {
 		result := captureStdout(func() {
-			l.UpgradeStarting("my-instance", "fake-guid")
+			l.UpgradeStarting(basicInstance("my-instance", "fake-guid"))
 		})
 		Expect(result).To(MatchRegexp(timestampRegexp + `: starting to upgrade instance: "my-instance" guid: "fake-guid"\n`))
 	})
 
 	It("can log the success of an upgrade", func() {
 		result := captureStdout(func() {
-			l.UpgradeSucceeded("my-instance", "fake-guid", time.Minute)
+			l.UpgradeSucceeded(basicInstance("my-instance", "fake-guid"), time.Minute)
 		})
 		Expect(result).To(MatchRegexp(timestampRegexp + `: finished upgrade of instance: "my-instance" guid: "fake-guid" successfully after 1m0s\n`))
 	})
 
 	It("can log the failure of an upgrade", func() {
 		result := captureStdout(func() {
-			l.UpgradeFailed("my-instance", "fake-guid", time.Minute, fmt.Errorf("boom"))
+			l.UpgradeFailed(basicInstance("my-instance", "fake-guid"), time.Minute, fmt.Errorf("boom"))
 		})
 		Expect(result).To(MatchRegexp(timestampRegexp + `: upgrade of instance: "my-instance" guid: "fake-guid" failed after 1m0s: boom\n`))
 	})
 
 	It("can log the final totals", func() {
 		l.InitialTotals(10, 5)
-		l.UpgradeFailed("my-first-instance", "fake-guid-1", time.Minute, fmt.Errorf("boom"))
-		l.UpgradeFailed("my-second-instance", "fake-guid-2", time.Minute, fmt.Errorf("bang"))
-		l.UpgradeSucceeded("my-third-instance", "fake-guid-3", time.Minute)
-		l.SkippingInstance("skipped", "skipped-guid", true, "create", "failed")
+		l.UpgradeFailed(basicInstance("my-first-instance", "fake-guid-1"), time.Minute, fmt.Errorf("boom"))
```

Comment:
nitpick: we are logging upgrade failed for an instance that doesn't have an upgrade available. If it is easy to pass the upgrade flag in true for these scenarios I would very much prefer that.

Reply:
Fixed by latest commit: https://github.com/cloudfoundry/upgrade-all-services-cli-plugin/pull/89/commits/daca87dd469c4ad258ae5da7f5d866bf2850c5f9